core/src/main/java/org/apache/iceberg/avro/Avro.java

```diff
@@ -217,17 +200,36 @@ static Context deleteContext(Map<String, String> config) {
         Context dataContext = dataContext(config);
 
         String codecAsString = config.get(DELETE_AVRO_COMPRESSION);
-        CodecFactory codec = codecAsString != null ? toCodec(codecAsString) : dataContext.codec();
+        CodecFactory codec = codecAsString != null ? toCodec(codecAsString, config) : dataContext.codec();
 
         return new Context(codec);
       }
 
-      private static CodecFactory toCodec(String codecAsString) {
+      private static CodecFactory toCodec(String codecAsString, Map<String, String> config) {
+        CodecFactory codecFactory = null;
         try {
-          return CodecName.valueOf(codecAsString.toUpperCase(Locale.ENGLISH)).get();
+          switch (Codec.valueOf(codecAsString.toUpperCase(Locale.ENGLISH))) {
+            case UNCOMPRESSED:
+              codecFactory = CodecFactory.nullCodec();
+              break;
+            case SNAPPY:
+              codecFactory = CodecFactory.snappyCodec();
+              break;
+            case ZSTD:
+              codecFactory = CodecFactory.zstandardCodec(PropertyUtil.propertyAsInt(config,
+                  AVRO_ZSTD_COMPRESSION_LEVEL,
+                  AVRO_ZSTD_COMPRESSION_LEVEL_DEFAULT));
+              break;
+            case GZIP:
+              codecFactory = CodecFactory.deflateCodec(9);
```

Comment:
Should we follow up with a way to set the deflate level as well? Maybe we don't care since it is such an old codec.

Reply:
yes with `write.delete.avro.compression-level` we should be able to set this using the config.

Reply:
Yeah with a common compression level key we can handle for GZIP as well. I still have set a default value for GZIP as 9.